File: Leetcode_submissions/problems/3sum_closest/solution.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        sort(nums.begin(), nums.end());
        int ans = nums[0] + nums[1] + nums[nums.size()-1];
        for (int i=0;i<nums.size()-2;i++){
            int l=i+1, r=nums.size()-1;
            int sum = nums[i]+nums[l]+nums[r];
            while (r > l){
                sum = nums[i] + nums[l] + nums[r];
                // cout << sum <<" " << i << " "<< l << " " <<r << endl;
                if (abs(target-sum) < abs(target-ans)){
                    ans = sum;
                }
                if (sum == target) return target;
                if (sum > target){
                    r--;
                }else{
                    l++;
                }
            }    
            // cout << sum << endl;
            
        }
        return ans;
        
    }
};

File: Leetcode_submissions/problems/3sum_closest/solution.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
    
        
        
        n = len(nums)
        nums.sort()
        ans = nums[0] + nums[1] + nums[2]
        
        for i in range(n-2):
            l, r = i+1, n-1
            
            while (r > l):
                cur = nums[i] + nums[l] + nums[r]
                
                if abs(target-cur) < abs(target-ans): ans = cur
                
                if cur == target: return target
                
                if cur > target: r -= 1
                else: l +=  1
        
        return ans
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        